### PR TITLE
Adding search to Groups API

### PR DESCRIPTION
**Pull request by ngaglani-tenable**
- Moved groups API to platform package.
     - Updated usage `tio.v3.platform.groups`
- Added support for search function.

### CODE REVIEW

**Review by dhvanilshah-tenable — APPROVED** [review]

File: tenable/io/v3/platform/groups/api.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        Returns:
+            List:
+                List of user resource records based on membership to the
+                specified group.
+
+        Examples:
+            >>> tio.v3.platform.groups.add_user(
+                    '00000000-0000-0000-0000-000000000000',
+                    '00000000-0000-0000-0000-000000000000'
+                )
```
Comment (harshshah-tenable) [inline tenable/io/v3/platform/groups/api.py]: Please add ... in examples at all instances

File: tenable/io/v3/platform/groups/api.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            group_id (UUID): The unique identifier of the group requested.
+
+        Returns:
+            List:
+                List of user resource records based on membership to the
+                specified group.
+
+        Example:
+            >>> group_id = '00000000-0000-0000-0000-000000000000'
+            >>> for user in tio.v3.platform.groups.list_users(group_id):
```
Comment (harshshah-tenable) [inline tenable/io/v3/platform/groups/api.py:172]: `>>>` -> ...

File: tenable/io/v3/platform/groups/api.py (hunk 1 above)
Comment (ngaglani-tenable) [reply tenable/io/v3/platform/groups/api.py]: Done.

**Review by harshshah-tenable — APPROVED** [review]

File: tenable/io/v3/platform/__init__.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+'''
+Platform
+========
+
+The following API's are available for interaction under Platform
+
+Methods available on ``tio.v3.was``:
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/__init__.py]: Is above statement correct, as it refers to **tio.v3.was** in platform module ?

File: tenable/io/v3/platform/groups/api.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+                List of user resource records based on membership to the
+                specified group.
+
+        Examples:
+            >>> tio.v3.platform.groups.add_user(
+            ...     '00000000-0000-0000-0000-000000000000',
+            ...     '00000000-0000-0000-0000-000000000000'
+            ... )
+        '''
+        self._post(f'{group_id}/users/{user_id}')
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/groups/api.py:57]: Add validation against proper value being passed for **group_id** and **user_id**.

File: tenable/io/v3/platform/groups/api.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                The name of the group that will be created.
+
+        Returns:
+            Dict:
+                The group resource record of the newly minted group.
+
+        Examples:
+            >>> group = tio.v3.platform.groups.create('Group Name')
+        '''
+        return self._post(json={'name': name})
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/groups/api.py:79]: Validate name against null/empty values, before forwarding it to server.

File: tenable/io/v3/platform/groups/api.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+        Returns:
+            None:
+                The group was successfully deleted.
+
+        Examples:
+            >>> tio.v3.platform.groups.delete(
+            ...     '00000000-0000-0000-0000-000000000000'
+            ... )
+        '''
+        self._delete(group_id)
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/groups/api.py:101]: **group_id** value should be validated, before being proxied in request to backend.

File: tenable/io/v3/platform/groups/api.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+            None:
+                The user was successfully removed from the group.
+
+        Examples:
+            >>> tio.v3.platform.groups.delete_user(
+            ...     '00000000-0000-0000-0000-000000000000',
+            ...     '00000000-0000-0000-0000-000000000000'
+            ... )
+        '''
+        self._delete(f'{group_id}/users/{user_id}')
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/groups/api.py:128]: Add validation here too.

File: tenable/io/v3/platform/groups/api.py [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+            Dict:
+                The group resource record.
+
+        Examples:
+            >>> tio.v3.platform.groups.edit(
+            ...     '00000000-0000-0000-0000-000000000000',
+            ...     'Updated name'
+            ... )
+        '''
+        return self._put(group_id, json={'name': name})
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/groups/api.py:154]: Add validation for name too.

File: tenable/io/v3/platform/groups/api.py [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+                As the filters may change and sortable fields may change over
+                time, it's highly recommended that you look at the output of
+                the :py:meth:`tio.v3.platform.definitions.groups()`
+                endpoint to get more details.
+            limit (int):
+                Number of objects to be returned in each request.
+                Default is 200.
+            next (str):
+                The pagination token to use when requesting the next page of
+                results.  This token is presented in the previous response.
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/groups/api.py]:
```suggestion
                results.  This token is obtained from the previous response.
```

File: tenable/io/v3/platform/groups/api.py [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+            return_resp (bool):
+                If set to true, will override the default behavior to return
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/groups/api.py]: Correct the docstring to match it with the actual behavior of the parameter.

File: tenable/io/v3/platform/groups/api.py [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+            return_csv (bool):
+                If set to true, It wil return the CSV Iterable. Returns all
+                data in text/csv format on each next call with row headers
+                on each page.
+        Returns:
+            Iterable:
+                The iterable that handles the pagination and potentially
+                async requests for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (Mangusingh-te) [inline tenable/io/v3/platform/groups/api.py]: Correct the docstring.

**Review by Mangusingh-te — CHANGES_REQUESTED** [review]

File: tenable/io/v3/platform/groups/api.py (hunk 4 above)
Comment (alpesh-te) [reply tenable/io/v3/platform/groups/api.py:57]: Added validation

File: tenable/io/v3/platform/groups/api.py (hunk 5 above)
Comment (alpesh-te) [reply tenable/io/v3/platform/groups/api.py:79]: Added validation

File: tenable/io/v3/platform/groups/api.py (hunk 6 above)
Comment (alpesh-te) [reply tenable/io/v3/platform/groups/api.py:101]: Added group_id validation

File: tenable/io/v3/platform/groups/api.py (hunk 7 above)
Comment (alpesh-te) [reply tenable/io/v3/platform/groups/api.py:128]: Added validation

File: tenable/io/v3/platform/groups/api.py (hunk 8 above)
Comment (alpesh-te) [reply tenable/io/v3/platform/groups/api.py:154]: Added validation

File: tenable/io/v3/platform/groups/api.py (hunk 11 above)
Comment (alpesh-te) [reply tenable/io/v3/platform/groups/api.py]: Addressed

File: tenable/io/v3/platform/groups/api.py (hunk 10 above)
Comment (alpesh-te) [reply tenable/io/v3/platform/groups/api.py]: addressed

**Review by alpesh-te — APPROVED** [review]

**Pull request merged**